Wrappers/Python/test/test_ring_processor.py

```diff
@@ -75,3 +77,43 @@ def test_L1Norm_2D(self):
         error = (ring_recon - sin).abs().as_array().mean()
         np.testing.assert_almost_equal(error, 83.20592, 4)
         
+class TestRingRemover(CCPiTestClass):
+    def add_defect_pixels(self, data, N_columns, N_defects, seed):
+
+        data_corrupted = data.copy()
+        rng = np.random.RandomState(seed=seed) 
+        columns = rng.randint(0, data.shape[1], size=N_columns)
+        
+        pixel_values = rng.uniform(low=np.amin(data.as_array()), high=np.amax(data.as_array()), size=N_defects)
+
+        for i in range(N_columns):
+            col_pattern = rng.uniform(low=np.amin(data.as_array()), high=np.amax(data.as_array()), size=data.shape[0])
+            data_corrupted.as_array()[:, columns[i]] = data.as_array()[:, columns[i]]+col_pattern
+
+        defect_row = rng.randint(0, data.shape[0], size=N_columns)
+        defect_col = rng.randint(0, data.shape[1], size=N_columns)
+        for i in range(N_defects):
+            data_corrupted.as_array()[defect_row[i], defect_col[i]] = pixel_values[i]
+
+        return data_corrupted
+    
+    def test_ring_remover(self):
+        
+        data = dataexample.SIMULATED_PARALLEL_BEAM_DATA.get().get_slice(vertical=80)
+
+        data_corrupted = self.add_defect_pixels(data, 1, 1, 1312)
+        data_corrupted = TransmissionAbsorptionConverter()(data_corrupted)
+        r = RingRemover(8,'db20', 1.5)
+        r.set_input(data_corrupted)
+        data_corrected = r.get_output()
+
+        self.assertEqual(data_corrupted.shape, data_corrected.shape)
+
+        data_corrupted_odd = Slicer(roi={'horizontal': (1, None)})(data_corrupted)
+        data_corrupted_odd = Slicer(roi={'angle': (1,None)})(data_corrupted_odd)
+        r = RingRemover(8,'db20', 1.5)
+        r.set_input(data_corrupted_odd)
+        data_corrected_odd = r.get_output()
+
+        self.assertEqual(data_corrupted_odd.shape, data_corrected_odd.shape)
```

Comment:
Add a test that checks the output rather than just the shape. You probably want it to be within a tolerance of the input image? Or find some metric that would work.

Reply:
Added a test for this